lib/spack/spack/test/projections.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+from datetime import date
+
+import spack.projections
+import spack.spec
+
+
+def test_projection_expansion(mock_packages):
+    """test that env variables and spack config variables are expanded in projections"""
+    os.environ["FOO_ENV_VAR"] = "test-string"
```

Comment:
Can you use `monkeypatch.setenv` here? Otherwise following tests will have a polluted environment.